expression/builtin_encryption_vec.go

```diff
@@ -35,11 +35,54 @@ import (
 )
 
 func (b *builtinAesDecryptSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinAesDecryptSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	strBuf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(strBuf)
+	if err := b.args[0].VecEvalString(b.ctx, input, strBuf); err != nil {
+		return err
+	}
+
+	keyBuf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(keyBuf)
+	if err := b.args[1].VecEvalString(b.ctx, input, keyBuf); err != nil {
+		return err
+	}
+
+	if b.modeName != "ecb" {
+		return errors.Errorf("unsupported block encryption mode - %v", b.modeName)
+	}
+
+	isWarning := !b.ivRequired && len(b.args) == 3
+
+	result.ReserveString(n)
+	for i := 0; i < n; i++ {
+		if strBuf.IsNull(i) || keyBuf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		if isWarning {
+			b.ctx.GetSessionVars().StmtCtx.AppendWarning(errWarnOptionIgnored.GenWithStackByArgs("IV"))
+		}
```

Comment:
We can separate this path out of this for loop.
```go
if isWarning {
    sctx := b.ctx.GetSessionVars().StmtCtx
    for {
       sctx.AppendWarning()
    }
}

for {
    ...
}
```

Reply:
we have to put the `appendWarning()` statement after the `if strBuf.IsNull(i) || keyBuf.IsNull(i){...}` statement so that the number of warnings in vectorized version will equal to the non-vectorized version.